packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts

```diff
@@ -109,65 +110,67 @@ export class CryptoWatchPriceFeed extends PriceFeedInterface {
     // If the time is before the first piece of data in the set, return null because
     // the price is before the lookback window.
     if (time < firstPricePeriod.openTime - this.historicalTimestampBuffer) {
-      throw new Error(`${this.uuid}: time ${time} is before firstPricePeriod.openTime`);
+      throw new Error(`${this.uuid}: time ${time} is before firstPricePeriod.openTime minus historicalTimestampBuffer`);
     }
 
     // historicalPricePeriods are ordered from oldest to newest.
     // This finds the first pricePeriod whose closeTime is after the provided time.
     const match = this.historicalPricePeriods.find((pricePeriod) => {
-      return (
-        time < pricePeriod.closeTime + this.historicalTimestampBuffer &&
-        time >= pricePeriod.openTime - this.historicalTimestampBuffer
-      );
+      return time <= pricePeriod.closeTime && time >= pricePeriod.openTime;
     });
 
+    // If match doesn't succeed, then we can still try to find the nearest price and use it if its within the caller's
+    // allowed margin of error.
+    let returnPrice;
     if (match === undefined) {
-      // If match doesn't succeed, then give caller details about closest price periods.
-      const before = this.historicalPricePeriods
-        .slice()
-        .reverse()
-        .find((pricePeriod) => time >= pricePeriod.openTime - this.historicalTimestampBuffer);
```

Comment:
I think this check was implemented incorrectly. To find the price period before `time`, we need to expand the period's `closeTime` not its `openTime`. To find the price period after `time` we should expand the `openTime`.